### PR TITLE
Update FRAM(ramtron) to 32-byte granularity

### PR DESCRIPTION
**Pull request by Igor-Misic**
This PR is one of the steps toward moving to LittleFS, but it is also helpful as a standalone since it improves BSON exporting time. 

It is dependent on NuttX PR: https://github.com/PX4/NuttX/pull/219 which needs to go in first.

#### Problem
RAMTRON driver uses a hardcoded value of 512 bytes for emulated pages and sectors. For LittleFS we are looking to use a smaller sector/page size with the best performance. 

#### Solution
Here is a graph showing different BSON files in sizes with the corresponding time required to export them to FRAM. 
In the graph, you can see how it looks performance when different granularity is used for the RAMTRON driver. 

image

It is clear that 32-byte granularity provides the best performance so here is a comparison with 512-byte granularity only. 

image

Even though we change the number of bytes that we are wringing and reading from FRAM the compatibility with existing parameters at FRAM should stay intact since the pattern is the same just with different sizes. 

Another thing that this PR brings is the removal of `/fs/mtd_waypoints` that is not in use for some long period. This is also preparation for LittleFS where we will be able to create a directory and write multiple files at full-size FRAM. 

### Test

Tested on fmu-v5x, fmu-v4 and cauv nora
image

Moving from 512-granularity to 32-byte granularity didn't corrupt the BSON file:
image

### CODE REVIEW

File: boards/ark/fmu-v6x/src/mtd.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #include <nuttx/spi/spi.h>
 #include <px4_platform_common/px4_manifest.h>
 //                                                              KiB BS    nB
-static const px4_mft_device_t spi5 = {             // FM25V02A on FMUM 32K 512 X 64
+static const px4_mft_device_t spi5 = {             // FM25V02A on FMUM 32K 32 X 1024
```
[the remark this answers is not in the document]
Comment (davids5) [reply boards/ark/fmu-v6x/src/mtd.cpp:37]: When looking up the FM25V02A none of the numbers make sense so maybe the above will be clearer. What do you think? If you think it helps replace the format it for all the part number.

[the remark this answers is not in the document]
Comment (Igor-Misic) [reply boards/ark/fmu-v6x/src/mtd.cpp:37]: huh, thanks for spotting this. I'll fix it.

**Comment by Igor-Misic** [comment]
> I see this is ALSO dropping waypoints. Does that make sense if we want to buffer them later?

that was a suggestion from @dagar. His comment is that it is not used for a very long time. 
Another thing is when we integrate LittleFS we are going to be able to write files like `/mtd_params/parameter.bson`, `/mtd_params/waypoint1`, `/mtd_params/waypoint_backup` + creating directories.  So we can already start thinking of renaming  `/mtd_params` to something more generic. For now, I don't have any idea about the new name.

**Review by davids5 — DISMISSED** [review]
LGTM,

**Comment by davids5** [comment]
@Igor-Misic https://github.com/PX4/NuttX/pull/219 was merged. Please repoint this to the NuttX tip. S we can bring it in.

**Comment by davids5** [comment]
@Igor-Misic  Please fetch NuttX and check out px4_firmware_nuttx-10.3.0+ pull it and ensure this PR is pointing to the tip pof px4_firmware_nuttx-10.3.0+

**Comment by davids5** [comment]
@dagar - What is up with HW CI?

**Comment by dagar** [comment]
> @dagar - What is up with HW CI?

It's a real PX4 failure someone slipped in that I haven't tracked down yet.

**Comment by dagar** [comment]
FYI I wanted to try making the switch to littefs right after we get the next stable release (v1.14) out or at least branches from main.

**Pull request merged**